### PR TITLE
refactor(conf): delete migrations for deprecated fields

### PR DESCRIPTION
**Pull request by thekip**
# Description

Remove migrations for deprecated options. 

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

Fixes # (issue)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: website/docs/releases/migration-4.md [hunk 1]
```diff
@@ -100,3 +100,13 @@ declare type ExtractorType = {
 ```
 
 Read more about custom extractor on the [Advanced: Custom Extractor](/guides/custom-extractor) page.
+
+### Configuration migrations for deprecated options were deleted.
```
Comment (Martin005) [inline website/docs/releases/migration-4.md:104]:
```suggestion
### Configuration migrations for deprecated options were deleted
```

File: website/docs/releases/migration-4.md [hunk 2]
```diff
@@ -100,3 +100,13 @@ declare type ExtractorType = {
...
+
+Migration for older
```
Comment (Martin005) [inline website/docs/releases/migration-4.md:106]:
```suggestion
Migration for the following older options:
```

File: website/docs/releases/migration-4.md [hunk 3]
```diff
@@ -100,3 +100,13 @@ declare type ExtractorType = {
...
+ - `LinguiConfig.localeDir`,
+ - `LinguiConfig.srcPathDirs`,
+ - `LinguiConfig.srcPathIgnorePatterns`
+ - `LinguiConfig.fallbackLocale`
+
+Were deleted from source code. This should affect only users who migrate v2 -> v4 directly.
```
Comment (Martin005) [inline website/docs/releases/migration-4.md:112]:
```suggestion
were deleted from the source code. This should affect only users who are migrating from `v2` to `v4` directly.
```

File: website/docs/releases/migration-4.md [hunk 4]
```diff
@@ -100,3 +100,13 @@ declare type ExtractorType = {
...
 
 Read more about custom extractor on the [Advanced: Custom Extractor](/guides/custom-extractor) page.
+
+### Configuration migrations for deprecated options were deleted.
+
+Migration for older
+ - `LinguiConfig.localeDir`,
+ - `LinguiConfig.srcPathDirs`,
+ - `LinguiConfig.srcPathIgnorePatterns`
+ - `LinguiConfig.fallbackLocale`
```
Comment (Martin005) [inline website/docs/releases/migration-4.md:110]:
```suggestion
 - `localeDir`,
 - `srcPathDirs`,
 - `srcPathIgnorePatterns`,
 - `fallbackLocale`
```

**Review by andrii-bodnar — APPROVED** [review]

**Comment by thekip** [comment]
@Martin005 done

**Review by Martin005 — APPROVED** [review]

**Pull request merged**